app/models/miq_queue_worker_base/runner.rb

```diff
@@ -74,7 +74,7 @@ def get_message_via_sql
     loop do
       msg = MiqQueue.get(
         :queue_name => @worker.queue_name,
-        :role       => @active_roles,
+        :role       => @worker.required_roles.presence,
```

Comment:
we should probably refactor the get part of this to a method so it's testable outside of the `loop do`

Reply:
:+1: